rust/main/submitter/src/chain_tx_adapter/error.rs

```diff
@@ -0,0 +1,29 @@
+#![allow(dead_code)]
+
+use hyperlane_base::db::DbError;
+
+use crate::transaction::Transaction;
+
+#[derive(Debug, thiserror::Error)]
+pub enum DispatcherError {
```

Comment:
I would call it `AdapterError` since they will be reported from adapter, as far as I understand it.

Reply:
agreed that of the two options below to first do 1), and aim to switch to 2) later
1) have a single error type across the entire Submitter. We rename the current enum to SubmitterError and move it to submitter/src to reduce confusion. The drawback here is that the errors for various components are mixed into one type
2) have component specific error types (AdapterError, DispatcherError), but also a submitter-wide SubmitterError which is an enum that combines AdapterError, DispatcherError. The drawback here is that some errors might need two conversion steps. For instance, ChainCommunicationError -> AdapterError(ChainCommunicationError) -> SubmitterError(AdapterError(ChainCommunicationError)). This would require us to use .map_err in many places, which is verbose and I would like to avoid